### PR TITLE
PE-1337: Data preparation

### PR DESCRIPTION
**Pull request by matibat**
**Adds methods for**
- Querying GQL
- Gathering the Staked PST holders

### CODE REVIEW

File: src/queries.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		console.log(`Error in query: \n${JSON.stringify(query, null, 4)}`);
+		throw new Error(errors.message);
+	}
+	return JSONBody.data.transactions as GQLTransactionsResultInterface;
+}
+
+function createQuery(minBlock: number, maxBlock: number): Query {
+	return {
+		query: `
+query {
```
Comment (javdhu) [inline src/queries.ts]: Might wanna fix the formatting here

Comment (matibat) [reply src/queries.ts]: e05546a9c969a6a371d9952f0ae9b8b4fb6dedd0

File: package.json [hunk 2]
```diff
@@ -31,6 +32,10 @@
 		"rimraf": "^3.0.2",
 		"typescript": "^4.6.3"
 	},
+	"dependencies": {
+		"@types/node-fetch": "^2.6.1",
```
Comment (karlprieb) [inline package.json]: types are dev dependencies

File: src/gql_Types.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
+// Arweave GraphQL Interfaces
```
Comment (karlprieb) [inline src/gql_Types.ts]: Why this file name has a capital `T` on `Types`?

Comment (matibat) [reply src/gql_Types.ts]: Copy/paste from CORE. I can rename it

File: package.json (hunk 2 above)
Comment (matibat) [reply package.json]: Nice catch
4fa1ab42efd63247db888cffa1d0ae5b6ec67980

**Review by karlprieb — COMMENTED** [review]
remove and add `.eslintcache` to `.gitignore` :)

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// import { getAllTransactionsWithin } from './queries';
+
+import { getWaleltsEligibleForStreak } from './queries';
+
+// const minBlock = 916705;
+// const maxBlock = 916756;
+
+if (require.main === module) {
```
Comment (karlprieb) [inline src/index.ts:8]: why do we need that?
can you also clean the comments?

File: src/queries.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+interface Query {
+	query: string;
+}
+
+export interface StakedPSTHolders {
+	[address: string]: number;
+}
+
+export async function getWaleltsEligibleForStreak(): Promise<StakedPSTHolders> {
```
Comment (karlprieb) [inline src/queries.ts]: why `async` if you're not using `await`?

File: src/queries.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+		.then((entries) => entries.filter((data) => data[1] >= 200))
+		.then((entries) => Object.fromEntries(entries));
+}
+
+async function getStakedPSTHolders(): Promise<StakedPSTHolders> {
+	// https://v2.cache.verto.exchange/-8A6RexFkpfWwuyVO98wzSFZh0d6VJuI-buTJvlwOJQ
+	// const result = await vertoClient.getExchangeDetails('ARDRIVE');
+	const blockHeightRequest = await fetch('https://arweave.net/height');
+	const blockHeight = +(await (await blockHeightRequest.blob()).text());
+	const result = await fetch('https://v2.cache.verto.exchange/-8A6RexFkpfWwuyVO98wzSFZh0d6VJuI-buTJvlwOJQ');
```
Comment (karlprieb) [inline src/queries.ts]: maybe a more meanful name, like `pstRequest`?
I was thinking now, why verto and not arweave gw? I know that we use verto on the price calculator. I'm just curious.

File: src/queries.ts [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+	const holders = await result.json();
+	const vault = holders.state.vault;
```
Comment (karlprieb) [inline src/queries.ts]:
```suggestion
	const { state: { vault } } = await result.json();
```

File: src/queries.ts [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+	const vaultAsArray = Object.entries(vault) as [string, Array<{ balance: number; start: number; end: number }>][];
+	const stakedForAtLeastOneMonth = vaultAsArray.map(([address, locks]) => [
+		address,
+		locks.reduce((accumulator, currentValue) => {
+			const start = currentValue.start;
+			const end = currentValue.end;
+			const lockedForAtLeastAMonth = blockHeight - start >= 21600;
```
Comment (karlprieb) [inline src/queries.ts]: maybe use a const for `21600`

File: .eslintcache [hunk 9]
```diff
@@ -0,0 +1 @@
+[{"/home/matibat/ardriveapp/inferno-backend/src/gql_types.ts":"1","/home/matibat/ardriveapp/inferno-backend/src/queries.ts":"2"},{"size":2674,"mtime":1650563489648,"results":"3","hashOfConfig":"4"},{"size":3635,"mtime":1650563489644,"results":"5","hashOfConfig":"4"},{"filePath":"6","messages":"7","suppressedMessages":"8","errorCount":0,"fatalErrorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"1fflav8",{"filePath":"9","messages":"10","suppressedMessages":"11","errorCount":0,"fatalErrorCount":0,"warningCount":0,"fixableErrorCount":0,"fixableWarningCount":0},"/home/matibat/ardriveapp/inferno-backend/src/gql_types.ts",[],[],"/home/matibat/ardriveapp/inferno-backend/src/queries.ts",[],[]]
```
Comment (matibat) [inline .eslintcache]: Cache? We should gitignore this

File: src/index.ts (hunk 4 above)
Comment (matibat) [reply src/index.ts:8]: There will be our main function call (unless we want this package to be a module) where we will read the argv and decide which method to run, maybe :)

File: src/queries.ts (hunk 7 above)
Comment (matibat) [reply src/queries.ts]: I might want to create a type for that :opionion:

File: src/gql_Types.ts (hunk 3 above)
Comment (matibat) [reply src/gql_Types.ts]: [91be1b2](https://github.com/ardriveapp/inferno-backend/pull/2/commits/91be1b293a22b7cd8affd0ea395d7bb21f309996)

File: src/queries.ts (hunk 5 above)
Comment (matibat) [reply src/queries.ts]: 510187db567602a4522f180595524524c23c4f7c

File: src/queries.ts (hunk 6 above)
Comment (matibat) [reply src/queries.ts]: 484898176e4e57673d28735fe5f928b9e44ec391

Comment (matibat) [reply src/queries.ts]: > why verto and not arweave gw?

Let's better discuss it with the team

File: .eslintcache (hunk 9 above)
Comment (matibat) [reply .eslintcache]: 9d821c61bdfd05c0e6f0025305df652419d42eb6

File: src/queries.ts (hunk 8 above)
Comment (matibat) [reply src/queries.ts]: [e6da8e6](https://github.com/ardriveapp/inferno-backend/pull/2/commits/e6da8e6c1dbf9c46e518fa9ad42b77aad6fe8c76)

File: src/queries.ts (hunk 7 above)
Comment (matibat) [reply src/queries.ts]: [e6da8e6](https://github.com/ardriveapp/inferno-backend/pull/2/commits/e6da8e6c1dbf9c46e518fa9ad42b77aad6fe8c76)

**Review by karlprieb — APPROVED** [review]

**Pull request merged**